src/NuGet.Server/Infrastructure/ServerPackageRepository.cs

```diff
@@ -685,6 +698,14 @@ private bool EnableDelisting
                 return _getSetting("enableDelisting", false);
             }
         }
+        private int RetentionRule
+        {
+            get
+            {
+                // If the setting is misconfigured, treat it as off (backwards compatibility).
+                return GetIntAppSetting("retentionRule", 0);
```

Comment:
We may need to validate the int value returned by the GetIntAppSetting method. Negative values should not be allowed and be treated as if the feature is disabled.